src/test/java/plugins/SshSlavesPluginTest.java

```diff
@@ -58,35 +55,74 @@
     }
 
     @Test public void connectWithPassword() {
-        SshSlaveLauncher launcher = slave.setLauncher(SshSlaveLauncher.class);
-        launcher.host.set(sshd.ipBound(22));
-        launcher.port(sshd.port(22));
-
-        SshCredentialDialog dia = launcher.addCredential();
-        UserPwdCredential cred = dia.select(UserPwdCredential.class);
-        cred.username.set("test");
-        cred.password.set("test");
-        cred.add();
+        configureDefaultSSHSlaveLauncher().pwdCredentials("test", "test");
         slave.save();
 
         verify();
     }
 
     @Test public void connectWithKey() {
-        SshSlaveLauncher launcher = slave.setLauncher(SshSlaveLauncher.class);
-        launcher.host.set(sshd.ipBound(22));
-        launcher.port(sshd.port(22));
-
-        SshCredentialDialog dia = launcher.addCredential();
-        SshPrivateKeyCredential cred = dia.select(SshPrivateKeyCredential.class);
-        cred.username.set("test");
-        cred.enterDirectly(sshd.getPrivateKeyString());
-        cred.add();
+        configureDefaultSSHSlaveLauncher().keyCredentials("test", sshd.getPrivateKeyString());
+        // To avoid test flakiness
+        elasticSleep(1000);
```

Comment:
Hmmm, I don't see why this path is different to using password credentials (which works without a sleep)?

The DOM would be empty if the UI has not populated the dropdown yet with the newly created item - so I get that but why it's only in this case is jumping out at me. 

If you do need to wait waiting for the element (credentials) to be visible would be a better option as that can shave time off the test (everything adds up!)